.github/workflows/perf-budgets.yml

```
@@ -0,0 +1,160 @@
+        if: ${{ !cancelled() }}
+        run: |
+          xvfb-run --auto-servernum node tests/tools/benchmarks/daemon-coldstart-bench.mjs \
+            --label ci \
+            --iterations 2 \
+            --output "$REPORT_DIR/daemon-coldstart.json"
+
+      # One step so a PR sees every budget violation at once rather than fixing
+      # startup, pushing, and only then learning idle CPU regressed too.
+      - name: Check perf budgets
+        if: ${{ !cancelled() }}
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:3:30: Double quote to prevent globbing and word splitting

(shellcheck)